prbt_hardware_support/test/unittests/unittest_filter_pipeline.cpp

```diff
@@ -21,36 +21,37 @@
 
 #include <prbt_hardware_support/filter_pipeline.h>
 
-namespace filter_pipeline_test
+namespace prbt_hardware_support
 {
 
-using namespace prbt_hardware_support;
-
-class FilterPipelineTest : public testing::Test
-{
-
-};
-
 /**
  * @brief Tests that exception is thrown if empty callback function is
  * specified.
  */
-TEST_F(FilterPipelineTest, testEmptyCallbackFunction)
-{
-  ros::NodeHandle nh;
-  FilterPipeline::TCallbackFunc cb;
-  EXPECT_THROW(FilterPipeline(nh, cb), std::invalid_argument);
+TEST(FilterPipelineTest, testEmptyCallbackFunction){
+  {
+    ros::NodeHandle nh;
+    FilterPipeline::TCallbackFunc cb;
+    EXPECT_THROW(FilterPipeline(nh, cb), std::invalid_argument);
+    nh.shutdown();
```

Comment:
Did you try without this line? `shutdown()` is called anyways during destruction of `nh`.

Reply:
I see that it makes a difference in the main, but not here.

Reply:
ok. lets try